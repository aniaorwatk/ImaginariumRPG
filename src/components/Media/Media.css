.media {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    gap:0 12px;
    width: 342px;
    margin: 20px;
}

.media__link {
    text-decoration: none;
    color: white;
}

.media__icon {
    display: flex;
    align-items: center;
    flex-direction: column;
}

.media__icon-img {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 60px;
    height: 60px;
    border-radius: 8px;
    background-color: rgba(59, 61, 81, 0.3);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

}

.media__icon-title {
    font-weight: 400;
    font-size: 12px;
    color: rgba(255, 255, 255, 0.8);
}