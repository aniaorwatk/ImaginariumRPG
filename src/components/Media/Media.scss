@import "./../scss/Main";

.media {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    gap:  12px;
    width: 342px;
    margin: 20px 0;



    .mediaLink {
        text-decoration: none;
        color: $color-white;

        &__icon {
            display: flex;
            align-items: center;
            flex-direction: column;

            &-img {
                display: flex;
                align-items: center;
                justify-content: center;
                width: 60px;
                height: 60px;
                border-radius: 8px;
                background-color: $color-dark-trans03;
                box-shadow: 0px 4px 4px $color-black-trans025;

                &:hover {
                    opacity: 0.85;
                }

                &--view {
                    width: 40px;
                    height: 40px;
                }
            }

            &-title {
                font-weight: 400;
                font-size: 12px;
                color: $color-white-trans08;
            }
        }
    }

}



@media only screen and (min-width: $breakpointM) {
    .media {
        gap: 24px;
        width: 469px;

        &__icon {

            &-img {
                width: 74px;
                height: 74px;

                &-view {
                    width: 54px;
                    height: 54px;
                }
            }

            &-title {
                font-size: 16px;
            }
        }
    }
}

@media only screen and (min-width: $breakpointL) {
    .media {
        width: 394px;

        &__icon {
            width: 74px;

            &-img {
                background:$color-bg-mediaL;
                border: 1px solid $color-border-mediaL;
                box-shadow: none;
            }
        }
    }
}
