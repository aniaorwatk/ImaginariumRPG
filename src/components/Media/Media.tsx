import { IMediaType } from "../../API/APIGet";
import "./Media.scss";

interface IAllMediaType {
    media: IMediaType[] | undefined
}

const Media = ({ media }: IAllMediaType) => {

    const allMedia = media ? media.map(one => {
        return (
            <a href={`${one.source}`} target="_blank" rel="noopener noreferrer" key={one.id} className="mediaLink">
                <div className="mediaLink__icon">
                    <div className="mediaLink__icon-img">
                        <img className="mediaLink__icon-img--view" src={one.filename} alt={one.filename} title={one.filename} loading="eager" width={200} height={200} />
                    </div>
                    <p className="mediaLink__icon-title">{one.title}</p>
                </div>
            </a>
        )
    }) : []

    return (
        media ? <div className="media">
            {allMedia}
        </div> : null
    )
}

export default Media
