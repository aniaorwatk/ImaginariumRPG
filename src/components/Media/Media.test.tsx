import { screen, render } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";
import Media from "./Media";

describe('Media Component', () => {
    let container: any = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    const obj = [
        {
            title: "Discord",
            filename: "Discord",
            id: 1,
            source: "Discord"
        }
    ]

    it('Should render media with data', async () => {

        const tree = render(<Media media={obj} />, container);

        const media = screen.getByText('Discord');

        expect(media).toBeInTheDocument();
    })

    it('Should render', async () => {

        const tree = render(<Media media={obj} />);

        expect(tree.baseElement).toMatchSnapshot()
    })
})
