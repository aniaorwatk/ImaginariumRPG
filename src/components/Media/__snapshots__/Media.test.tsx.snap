// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Component Should render 1`] = `
<body>
  <div />
  <div>
    <div
      class="media"
    >
      <a
        class="mediaLink"
        href="Discord"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="mediaLink__icon"
        >
          <div
            class="mediaLink__icon-img"
          >
            <img
              alt="Discord"
              class="mediaLink__icon-img--view"
              height="200"
              loading="eager"
              src="Discord"
              title="Discord"
              width="200"
            />
          </div>
          <p
            class="mediaLink__icon-title"
          >
            Discord
          </p>
        </div>
      </a>
    </div>
  </div>
</body>
`;
